src/Compilers/Core/Portable/Symbols/IMethodSymbol.cs

```diff
@@ -90,13 +90,24 @@ public interface IMethodSymbol : ISymbol
         /// </summary>
         ITypeSymbol ReturnType { get; }
 
+        /// <summary>
+        /// Gets the declared nullability of the return type of the method.
+        /// </summary>
+        Nullability ReturnNullability { get; }
+
         /// <summary>
         /// Returns the type arguments that have been substituted for the type parameters. 
         /// If nothing has been substituted for a given type parameter,
         /// then the type parameter itself is consider the type argument.
         /// </summary>
         ImmutableArray<ITypeSymbol> TypeArguments { get; }
 
+        /// <summary>
+        /// Returns the nullability of the type arguments that have been substituted for the
+        /// type paramters.
+        /// </summary>
+        ImmutableArray<Nullability> TypeArgumentsNullability { get; }
```

Comment:
this makes me like things even less.  Note: the naming here is good (though the naming of ReturnNullability is inconsistent).  But it seems pretty unfortunate that it's exposed through an entirely separate array, instead of being exposed on the individual TypeArgument type.